Database.cs

```diff
@@ -0,0 +1,102 @@
+﻿using MongoDB.Driver;
+
+namespace InventoryManagmentMongoDB
+{
+    public class Database
+    {
+        private readonly MongoDbContext _context;
+
+        public Database(string connectionString, string databaseName)
+        {
+            _context = new MongoDbContext(connectionString, databaseName);
+        }
+
+        public void AddProduct(Product product)
+        {
+            var existingProduct = _context.Products.Find(p => p.Name == product.Name).FirstOrDefault();
+            if (existingProduct != null)
+            {
+                throw new System.Exception("A product with the same name already exists.");
+            }
+            _context.Products.InsertOne(product);
+        }
+
+        public List<Product> GetAllProducts()
+        {
+            return _context.Products.Find(p => true).ToList();
+        }
+
+        public Product FindProductByName(string name)
+        {
+            return _context.Products.Find(p => p.Name == name).FirstOrDefault();
+        }
+
+        public Product FindProductById(string productId)
```

Comment:
we need to ensure that all database operations are performed asynchronously to improve performance and responsiveness. so plz update all methods interacting with the database or indirectly invoking database operations to be async.

so for this method it should be like this

`  public async Task<Product> FindProductByNameAsync(string name)
  {
      return _context.Products.Find(p => p.Name == name).FirstOrDefaultAsync();
  }`

**Note method name**